### ISSUE TITLE
Again lost with git

### ISSUE THREAD

**Issue by stefaweb** [mailwatch/1.2.0#186]
Sorry, I'm lost again with Git.

I don't know if all theses commits are well deployed and if you see them.

![capture d ecran 2015-06-19 a 14 43 02](https://cloud.githubusercontent.com/assets/12907102/8253598/ba2994a8-1691-11e5-8844-08f69c70a31f.png)

Seem not.

**Comment by endelwar** [comment]
mmmm... did you do an hard reset before committing to your branch?

**Comment by stefaweb** [comment]
Still lost.
My local and fork are in sync.
But I can see all these diff between mailwatch/1.2.0 and stefaweb/1.2.0

![capture d ecran 2015-06-19 a 20 27 47](https://cloud.githubusercontent.com/assets/12907102/8260140/e9945306-16c1-11e5-8d60-23f031446faf.png)

I don't understand if my commits are hold due to an error from me OR if these commits are only waiting for you to be merged?

**Comment by endelwar** [comment]
I still have to merge them, but I can't merge in this state, there are duplicated commit (I should not see my commit in your branch)

**Comment by stefaweb** [comment]
Hello!

Can I do something on my side or this problem should be solved by you?

**Comment by stefaweb** [comment]
ok, I have created a new branch (Update upgrade master), but not pulled yet.

I have done git fetch upstream && git reset --hard upstream/master locally and I now see my local master branch in sync with the mailwatch/master (all my mods are trashed).

All my mods are still in  stefaweb:update-upgrade-master

But when I tried to push my local master to my master branch, I got this.

But

[MacMiam:~/GitHub/1.2.0] sleclerc% git push
warning: push.default is unset; its implicit value has changed in
Git 2.0 from 'matching' to 'simple'. To squelch this message
and maintain the traditional behavior, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

When push.default is set to 'matching', git will push local branches
to the remote branches that already exist with the same name.

Since Git 2.0, Git defaults to the more conservative 'simple'
behavior, which only pushes the current branch to the corresponding
remote branch that 'git pull' uses to update the current branch.

See 'git help config' and search for 'push.default' for further information.
(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode
'current' instead of 'simple' if you sometimes use older versions of Git)

To https://github.com/stefaweb/1.2.0.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/stefaweb/1.2.0.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
